pages/release-process.md

```diff
@@ -1,111 +1,113 @@
-
- 5. Remove all RC tags
-
-    ```shell
-    # Remove tag locally
-    git tag -d 1.0.0-beta-1-RC1
-    # Remove tag on fork
-    git push --delete origin 1.0.0-beta-1-RC1
-    ```
- 6.  View the [website README] for instructions on how to generate Javadocs and documentation using the released tag.  Submit PR to the website repo to publish.
- 
- 7.  Send an email to `dev@fluo.incubator.apache.org` announcing new release.
-
-[website README]: https://github.com/apache/incubator-fluo-website/blob/master/README.md
-[documentation]: http://central.sonatype.org/pages/apache-maven.html
-[sonatype account]: https://issues.sonatype.org/
-[Release the artifacts]: http://central.sonatype.org/pages/releasing-the-deployment.html
+---
+layout: page
+title: Release Process
+permalink: /release-process/
+---
+
+### Initial Setup
+
+Before you can release Fluo or Fluo Recipes, you will need a GPG key. For information on generating
+a key look at this [ASF GPG page](https://www.apache.org/dev/openpgp.html).  After generating a key,
+add it to the [KEYS](https://www.apache.org/dist/incubator/fluo/KEYS) file.
+
+The maven release plugin will need credentials to stage artifacts.  You can provide the credentials
+by adding the following to your `~/.m2/settings.xml` file.  You may want remove your password after
+running maven.
+
+```xml
+<servers>
+  <server>
+    <id>apache.releases.https</id>
+    <username>your-apache-id</username>
+    <password>your-apache-password</password>
+  </server>
+</servers>
+```
+
+### Release Fluo
+
+Before starting the release process below, the following tasks should be complete:
+
+ * Create release notes for project website using GitHub issues.
+ * Perform testing and document results
+ * Start a gpg-agent to cache your gpg key to avoid entering your passphrase multiple times.
+
+   ```shell
+   gpg-connect-agent reloadagent /bye
+   ```
+
+Next, repeat the steps below until a good release candidate (RC) is found.  The script in
+`contrib/create-release-candidate.sh` automates this process.  However there is no guarantee that it
+works correctly.  Before using the script ensure you understand the process and inspect the script.
+In the following steps `RCV` is short for release candidate version.
+
+ 1. Branch master (or the current snapshot) and call the branch `<releaseVersion>-rc<RCV>-next`
+
+ 2. Except in Maven POMs (which will be updated by the next step), change any version references in docs and code from `<releaseVersion>-SNAPSHOT` to `<releaseVersion>`.  Commit changes to branch.
+
+ 3. Prepare the release which will verify that all tests pass: `mvn release:prepare`
+
+ 4. Perform the release: `mvn release:perform`
+    * This step will create a staging repository viewable only to you when login to https://repository.apache.org
+    * When `release:perform` finishes, you will need to close the staging repository to make it viewable to anyone at  https://repository.apache.org/content/repositories/orgapachefluo-REPO_ID
```

Comment:
It's important that the staging repo is *never* promoted/released until after a vote is successful, because that will push its contents to Maven central as though it were an official release. This can't be undone, and the only correct course of action (if the vote fails) is to skip over that release number for the next release candidate.